exporters/otlp/common/src/test/java/io/opentelemetry/exporter/internal/retry/RetryInterceptorTest.java

```diff
@@ -150,10 +149,7 @@ void connectTimeout() throws Exception {
     assertThatThrownBy(
             () ->
                 client.newCall(new Request.Builder().url("http://10.255.255.1").build()).execute())
-        .isInstanceOf(SocketTimeoutException.class)
-        .matches(
-            (Predicate<Throwable>)
-                throwable -> throwable.getMessage().equalsIgnoreCase("connect timed out"));
```

Comment:
Weird this was only null sporadically 🤷

Reply:
The way I understand it is that there are 2 places where `SocketTimeoutException` is thrown. `SocketTimeoutException` with null message is thrown when timeout is reached before it actually attempts to connect. `SocketTimeoutException` with message is thrown a bit later. Which you get depends on how fast the test is executed.